src/nccl_ofi_cuda.cpp

```diff
@@ -75,13 +79,45 @@ DECLARE_CUDA_FUNCTION(cuMemcpy, 4000);
 
 int nccl_net_ofi_cuda_init(void)
 {
+	int ret = 0;
+	void *cudaruntime_lib = NULL;
+	char libcudart_path[1024];
 	int driverVersion = -1;
 	int runtimeVersion = -1;
+	cudaError_t res;
+	CUresult cu_ret;
+	
+	/* Load CUDA runtime library only for functions without driver equivalents */
+	snprintf(libcudart_path, 1024, "%s", "libcudart.so");
+
+	(void) dlerror(); /* Clear any previous errors */
+	cudaruntime_lib = dlopen(libcudart_path, RTLD_NOW);
```

Comment:
This is not what we talked about.  The default should be to use cudart_static.  We only want to change behavior when the static library support is disabled (ie, our binaries).